### DISCUSSION TITLE
how to Implement I/O in OpENer  c++ raspberrypi

### DISCUSSION THREAD

**Discussion by balaji-6975** [EIPStackGroup/OpENer#367] [Q&A] [answered]
HI 👍 
I have downloaded&installed OpENer package in raspberrypi with the guidance of  Readme file in Package .
after that, I tried to connect plc(with given eds file) in command line , it give connection sccuessfull in plc .but now Iam not able to communicate with plc (with eds file) by sending and receiving data .
if I/O communication is possible then how to implement it and please give some sample c++ code related to connection and communication in OpENer  
 
regards 
Balaji Guru

![2022-03-21-154219_1920x1080_scrot](https://user-images.githubusercontent.com/59054762/159241806-b8aa6999-7abb-40f5-a115-22bda72adcda.png)

**Comment by MartinMelikMerkumians** [comment, marked as answer]
Hi.
OpENer implements IO (class 1) communication. It is also an adapter device (server), so it can't start communication with a PLC, nor is it intended to do so.
You can exchange data with the assemblies, in fact the input assembly can be read by the PLC.

PS: You can't see any output as the traces are probably disabled
